c_glib/arrow-glib/array.cpp

```diff
@@ -609,6 +609,24 @@ garrow_int8_array_get_value(GArrowInt8Array *array,
   return static_cast<arrow::Int8Array *>(arrow_array.get())->Value(i);
 }
 
+/**
+ * garrow_int8_array_get_values:
+ * @array: A #GArrowInt8Array.
+ * @length: (out): The number of values.
+ *
+ * Returns: (array length=length): The raw values.
+ */
+const gint8 *
+garrow_int8_array_get_values(GArrowInt8Array *array,
+                             gint64 *length)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto arrow_int8_array =
+    std::static_pointer_cast<arrow::Int8Array>(arrow_array);
+  *length = arrow_int8_array->length();
+  return arrow_int8_array->raw_data();
```

Comment:
Seems like this could be a single template:

`return GetValues<Int8Type>(array)`

You can use `TypeTraits<T>::ArrayType`